parser/auth/sm3.go

```diff
@@ -0,0 +1,384 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash"
+	"strconv"
+)
+
+// The concrete SM3 Cryptographic Hash Algorithm can be accessed in http://www.sca.gov.cn/sca/xwdt/2010-12/17/content_1002389.shtml
+// This implementation of 'type sm3 struct' is modified from https://github.com/tjfoc/gmsm/tree/601ddb090dcf53d7951cc4dcc66276e2b817837c/sm3
+// Some other references:
+// 	https://tools.ietf.org/id/draft-oscca-cfrg-sm3-01.html
```

Comment:
As you can see on https://datatracker.ietf.org/doc/draft-oscca-cfrg-sm3/ there is a v02 and it is also replace by 'draft-sca-crg-sm3'. The link also doesn't work for me unless I chop of the '.html'.

```suggestion
// 	https://datatracker.ietf.org/doc/draft-sca-cfrg-sm3/
```